doc/globbing.rdoc

```diff
@@ -0,0 +1,356 @@
+    # => [#<Pathname:/>, #<Pathname:basictest/>, #<Pathname:benchmark/>]
+
+    Dir.glob(pattern).reverse.take(5)
+    # => ["yjit/src/backend/x86_64/", "yjit/src/backend/arm64/", "yjit/src/backend/", "yjit/src/asm/x86_64/", "yjit/src/asm/arm64/inst/"]
+    Pathname.glob(pattern).reverse.take(3)
+    # => [#<Pathname:yjit/src/backend/x86_64/>, #<Pathname:yjit/src/backend/arm64/>, #<Pathname:yjit/src/backend/>]
+    pathname.glob(pattern).reverse.take(2)
+    # => [#<Pathname:yjit/src/backend/x86_64/>, #<Pathname:yjit/src/backend/arm64/>]
+
+  If the string pattern contains other characters
+  or is not followed by a slash character, it is equivalent to <tt>'*'</tt>.
+
+- <tt>'?'</tt>: matches any single character.
+  Equivalent to regexp <tt>/./</tt>:
+
+    pattern = 'io.?'
+    Dir.glob(pattern)      # => ["io.c"]
+    Pathname.glob(pattern) # => [#<Pathname:io.c>]
+    pathname.glob(pattern) # => [#<Pathname:io.c>]
+
+- <tt>'[_characters_]'</tt>: matches any single character in _characters_;
+  behaves like a \Regexp {character class}[rdoc-ref:Regexp#Character+Classes],
+  including set negation (e.g., <tt>'[^a-z]'</tt>):
+
+    pattern = '*.[a-z][a-z]'
+    Dir.glob(pattern).take(5)
+    # => ["array.rb", "ast.rb", "autogen.sh", "common.mk", "configure.ac"]
+    Pathname.glob(pattern).take(3)    # => [#<Pathname:array.rb>, #<Pathname:ast.rb>, #<Pathname:autogen.sh>]
+    pathname.glob(pattern).take(3)
+    # => [#<Pathname:array.rb>, #<Pathname:ast.rb>, #<Pathname:autogen.sh>]
+
+- <tt>'\'</tt>: escapes the next metacharacter:
+
+  Note that on Windows, the backslash character may not be used in a string pattern:
+  <tt>Dir['c:\foo*']</tt> will not work, use <tt>Dir['c:/foo*']</tt> instead.
+
+See also flag File::FNM_EXTGLOB below.
+
+=== Flags
+
+\Integer _flags_ may be used to restrict, expand, or otherwise modify
+the behavior of the filename globbing methods:
+
+- Dir.glob and Pathname.glob accept optional keyword argument +flags+
+  with an integer value:
+
+    Dir.glob('*', flags: File::FNM_DOTMATCH).take(5)
+    # => [".", ".appveyor.yml", ".cirrus.yml", ".dir-locals.el", ".document"]
+    Pathname.glob('*', flags: File::FNM_DOTMATCH).take(3)
+    # => [#<Pathname:.>, #<Pathname:.appveyor.yml>, #<Pathname:.cirrus.yml>]
+
+- Pathname#glob accepts an optional second argument as a flags value:
+
+    pathname.glob('*', File::FNM_DOTMATCH).take(3)
+    # => [#<Pathname:.>, #<Pathname:.appveyor.yml>, #<Pathname:.cirrus.yml>]
+
+The flags value may be the bitwise OR of any combination of individual
+flag constants;  example:
+
+  flags = File::FNM_CASEFOLD | File::FNM_DOTMATCH
+
+==== Flag File::FNM_CASEFOLD
+
+The FNM_CASEFOLD flag specifies that the matching is to be case-insensitive:
+
+  Dir.glob('c*').take(5)
+  # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja", "ccan", "class.c"]
+  Dir.glob('c*', flags: File::FNM_CASEFOLD).take(5)
+  # => ["CONTRIBUTING.md", "COPYING", "COPYING.ja", "ccan", "class.c"]
```

Comment:
`FNM_CASEFOLD` is only useful for `File.fnmatch`, and not for any other methods (including `Dir.glob`). See the [documentation](https://docs.ruby-lang.org/en/master/File/Constants.html).